### PR TITLE
Implement related videos

### PR DESCRIPTION
**Pull request by collection-man-globerskin**
This branch is a sub-branch of the ShowOptionsList

**On This PR**

- Table view for related videos implemented
- A view with a label for the title of the video
- Quality options for related videos appears when a cell is tapped (as an action sheet)
- A video is played when an option is selected

ezgif com-video-to-gif (3)

### CODE REVIEW

**Comment by collection-man-globerskin** [comment]
Support landscape and portrait for video player


Screen Shot 2019-10-22 at 11 48 36
Screen Shot 2019-10-22 at 11 48 54

File: VideoStreaming/StreamPlayerViewController.swift [hunk 1]
```diff
@@ -11,38 +11,159 @@ import AVKit
...
     
     required init?(coder: NSCoder) {
-        streamUrl = URL(fileURLWithPath: "")
-        super.init(coder: coder)
+        fatalError("init(coder:) has not been implemented")
     }
     
-    func play() {
-        player = AVPlayer(url: streamUrl)
+    override func viewDidLoad() {
```
Comment (ja-gutierrez) [inline VideoStreaming/StreamPlayerViewController.swift:55]: faltó llamar al super, ;). Esto puede que haya afectado en algo, jeje.

File: VideoStreaming/StreamPlayerViewController.swift [hunk 2]
```diff
@@ -11,38 +11,159 @@ import AVKit
...
     
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+
+        if landscapeVideoView == nil {
+            landscapeVideoView  = UIView(frame: CGRect(x: 0, y: 0, width: view.frame.height, height: view.frame.width))
+            view.addSubview(landscapeVideoView)
+            playerViewController.view.frame = landscapeVideoView.bounds
+            landscapeVideoView.addSubview(playerViewController.view)
+            landscapeVideoView.willMove(toSuperview: view)
```
Comment (ja-gutierrez) [inline VideoStreaming/StreamPlayerViewController.swift]: Para que estás usando este método?

**Review by ja-gutierrez — CHANGES_REQUESTED** [review]

File: VideoStreaming/StreamPlayerViewController.swift (hunk 1 above)
Comment (collection-man-globerskin) [reply VideoStreaming/StreamPlayerViewController.swift:55]: Agregado

File: VideoStreaming/StreamPlayerViewController.swift (hunk 2 above)
Comment (collection-man-globerskin) [reply VideoStreaming/StreamPlayerViewController.swift]: Al final, no hacía falta llamar a ese método. Ya lo quite

File: VideoStreaming/SubViews/Cells/VideoTableViewCell.swift [hunk 3]
```diff
@@ -17,7 +17,12 @@ import UIKit
         static let placeholderImg = "Placeholder"
     }
     
-    func configure(with video: Stream) {
+    func configure(with stream: Stream) {
+        videoView.configure(with: stream)
+        videoView.widthAnchor.constraint(equalTo: self.widthAnchor).isActive = true
```
Comment (ja-gutierrez) [inline VideoStreaming/SubViews/Cells/VideoTableViewCell.swift:22]: Jaja, me sigue sin cuadrar esto, pero bueno, jaja. Pa un futuro.

**Review by ja-gutierrez — APPROVED** [review]

**Review by castadev85 — APPROVED** [review]

**Pull request merged**